### PR TITLE
Added resource & data source for object store credentials

### PR DESCRIPTION
**Pull request by satakshigarg**
Added Terraform resources and data sources for object store credentials and their test files.

Signed-off-by: satakshigarg <satakshi@civo.com>

### CODE REVIEW

**Comment by alejandrojnm** [comment]
LGTM but we need test this, we need wait for this feature in production or staging or have the same in the KubeFake

**Comment by satakshigarg** [comment]
> LGTM but we need test this, we need wait for this feature in production or staging or have the same in the KubeFake

Yes @alejandrojnm. Testing is still pending. I and @haardikdharma10 will test it with your assistance.

File: civo/datasource_object_store.go [hunk 1]
```diff
@@ -38,28 +38,18 @@ func dataSourceObjectStore() *schema.Resource {
...
-				Description: "The maximum size of the Object Store",
-			},
-			"access_key_id": {
-				Type:        schema.TypeString,
 				Computed:    true,
-				Description: "The access key id of the Object Store",
+				Description: "The maximum size of the Object Store",
 			},
-			"secret_access_key": {
+			"credential_id": {
```
Comment (RealHarshThakur) [inline civo/datasource_object_store.go:47]: This should refer to access key ID instead.

File: civo/datasource_object_store.go [hunk 2]
```diff
@@ -103,18 +93,12 @@ func dataSourceObjectStoreRead(ctx context.Context, d *schema.ResourceData, m in
...
 	d.SetId(foundStore.ID)
 	d.Set("name", foundStore.Name)
-	d.Set("generated_name", foundStore.GeneratedName)
-	d.Set("max_size_gb", maxSize)
-	d.Set("access_key_id", foundStore.AccessKeyID)
-	d.Set("secret_access_key", foundStore.SecretAccessKey)
-	d.Set("endpoint", foundStore.ObjectStoreEndpoint)
+	d.Set("region", apiClient.Region)
+	d.Set("max_size_gb", foundStore.MaxSize)
+	d.Set("credential_id", foundStore.OwnerInfo.CredentialID)
```
Comment (RealHarshThakur) [inline civo/datasource_object_store.go:100]: access key ID here

File: civo/datasource_object_store_credentials.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The secret access key of the Object Store Credential",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the Object Store Credential",
+			},
+			"suspended": {
```
Comment (RealHarshThakur) [inline civo/datasource_object_store_credentials.go:56]: This shoudln't be exposed to users, it's not in the API either anymore

File: civo/resource_object_store_credentials.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+				Computed:    true,
+				Description: "The access key id of the Object Store Credential. It is generated by the provider.",
+			},
+			"secret_access_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The secret access key of the Object Store Credential. It is generated by the provider.",
+			},
+			"suspended": {
```
Comment (RealHarshThakur) [inline civo/resource_object_store_credentials.go:45]: This shouldn't be here

Comment (alejandrojnm) [reply civo/resource_object_store_credentials.go:45]: Done

File: civo/datasource_object_store.go [hunk 5]
```diff
@@ -6,7 +6,7 @@ import (
 	"strings"
 
 	"github.com/civo/civogo"
-	"github.com/civo/terraform-provider-civo/internal/utils"
+	// "github.com/civo/terraform-provider-civo/internal/utils"
```
Comment (RealHarshThakur) [inline civo/datasource_object_store.go:9]: We can remove this?

**Review by RealHarshThakur — APPROVED** [review]

**Pull request merged**